Auth0/CredentialsManager.swift

```diff
@@ -5,10 +5,36 @@ import JWTDecode
 import LocalAuthentication
 #endif
 
+/// Generic storage API for storing credentials
+public protocol CredentialsStorage {
+    /// Retrieve a storage entry
+    ///
+    /// - Parameters:
+    ///   - forKey: The key to get from the store
+    /// - Returns: The stored data
+    func data(forKey: String) -> Data?
```

Comment:
Following the [API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/), the methods can be renamed as follows:

```swift
func getEntry(forKey: String) -> Data?
func setEntry(_: Data, forKey: String) -> Bool
func deleteEntry(forKey: String) -> Bool
```

Usage:

```swift
guard let data = storage.getEntry(forKey: z) else { return }  // get entry for key z
guard storage.setEntry(y, forKey: z) else { return } //  set entry y for key z
guard storage.deleteEntry(forKey: z) else { return }  // delete entry for key z
```

> Include all the words needed to avoid ambiguity for a person reading code where the name is used.

> Compensate for weak type information to clarify a parameter’s role.
Especially when a parameter type is NSObject, Any, AnyObject, or a fundamental type such Int or String, type information and context at the point of use may not fully convey intent.
To restore clarity, precede each weakly typed parameter with a noun describing its role

> Prefer method and function names that make use sites form grammatical English phrases.

> Name functions and methods according to their side-effects
Those with side-effects should read as imperative verb phrases

That will require wrapping the SimpleKeychain methods in the extension.